lib/archethic_web/controllers/api/origin_key_controller.ex

```diff
@@ -80,12 +94,18 @@ defmodule ArchethicWeb.API.OriginKeyController do
 
   defp handle_error(error) do
     case error do
-      er when er in [:error, false] ->
+      false ->
         {400, %{status: "error - invalid public key"}}
 
       {:exists?, true} ->
         {400, %{status: "error - public key exists"}}
 
+      :error ->
+        {400, %{status: "error - certificate error"}}
+
+      {:error, :invalid_certificate} ->
```

Comment:
How do you distinguish between invalid hex from public key and certificate ?
The easiest would be to leverage Ecto.Changeset with a schema to allow public key in hex while ensuring it's a valid public key and certificate which should be hex based as well.